### DISCUSSION TITLE
Custom error message with custom matchers

### DISCUSSION THREAD

**Discussion by Linuus** [vcr/vcr#877] [Q&A]
Hi!

When I add the `:headers` matcher, the error outputs all headers. Now, I've created a custom matcher which only checks the `Authorization` header. If that fails, it just outputs the uri that doesn't match.

If my basic auth matcher fails, can I somehow output that information in the error message?

**Comment by krainboltgreene** [comment]
You can catch and reraise to your heart's content, it's Ruby.

**Comment by esposito** [comment]
What would be a good place to catch and reraise? Would be nice to define this logic in one place.

**Comment by esposito** [comment]
Solved it now using this monkey patch:

```ruby
module UnhandledHTTPRequestErrorDescriptionHelpers
  def match_request_on_headers?
    current_matchers.include?(:headers) || current_matchers.any? { |matcher| matcher.to_s.start_with?("header_") }
  end

  def match_request_on_body?
    current_matchers.include?(:body) || current_matchers.any? { |matcher| matcher.to_s.start_with?("body_") }
  end
end

module VCR
  module Errors
    class UnhandledHTTPRequestError
      prepend UnhandledHTTPRequestErrorDescriptionHelpers
    end
  end
end
```
